### DISCUSSION TITLE
Handling many preview environments

### DISCUSSION THREAD

**Discussion by Aaronius** [Unleash/unleash#3816] [Q&A] [answered]
Hello! Is there a recommended way to handle many preview environments? For example, if we have one preview environment for every pull request being made (so a change can be previewed before being merged), that could be a lot manage within Unleash.

Perhaps there's a mechanism for overriding flags using a query string or something like that that could assist us in flipping flags back and forth in these types of environments?

**Comment by markunl** [comment, marked as answer]
Hello Aaron, thanks for reaching out.

Yes, I believe constraints could help in this type of scenario, paired with an [Activation Strategy](https://docs.getunleash.io/reference/strategy-constraints).

Unleash context would limit the scope of the strategy to only those clients that match those parameters. Some built in ones include targeting users from specific regions, email addresses, user IDs, specific time windows ([docs](https://docs.getunleash.io/reference/unleash-context#structure)). 

These can be expanded using [custom context fields](https://docs.getunleash.io/how-to/how-to-define-custom-context-fields) to add anything else missing that your application is aware of - for example, the preview environment name.

Happy to jump on a call to discuss further, just let us know!

Thanks,


Mark
